chain-abci/src/app/validate_tx.rs

```diff
@@ -80,63 +78,181 @@ impl<T: EnclaveProxy> ChainNodeApp<T> {
         };
         match txaux {
             TxAux::EnclaveTx(tx) => {
-                let r = verify_enclave_tx(
+                let action = verify_enclave_tx(
                     &mut self.tx_validator,
                     &tx,
                     extra_info,
                     &self.uncommitted_account_root_hash,
                     &self.storage,
                     &self.accounts,
-                );
-                match (r, store_valid) {
-                    (Err(e), _) => Err(e),
-                    (Ok((fee, acc, _)), false) => Ok((fee, acc)),
-                    (Ok((fee, acc, None)), _) => Ok((fee, acc)),
-                    (Ok((fee, acc, Some(log))), true) => {
-                        self.storage.store_sealed_log(&txaux.tx_id(), &log);
-                        Ok((fee, acc))
-                    }
-                }
+                )?;
+                Ok(TxAction::Enclave(action))
             }
-            _ => verify_public_tx(
-                &txaux,
-                extra_info,
-                state,
-                &self.uncommitted_account_root_hash,
-                &self.accounts,
-            ),
+            _ => Ok(TxAction::Public(min_fee)),
```

Comment:
Yes, currently part of the logic is a bit in flux, the only purpose of `Public` branch here, is to avoid re-computing the `min_fee` at public tx processing, I'll add a comment here.

What do you think of distinguishing the enclave tx and public tx in `Tx/TxAux` level, like:
```
pub enum TxAux {
    EnclaveTx(TxEnclaveAux),
    PublicTx(TxPublicAux),
}
```
Then we'll have something to represent public tx only, avoid some panicking in tx pattern matching when processing public tx only.

Reply:
Yes, I think for the transaction wire format stabilisation, it'll be good to group those transaction types

Reply:
that can be in a separate pr

Reply:
https://github.com/crypto-com/chain/issues/1222